### PR TITLE
Add ResNeXt

### PR DESCRIPTION
**Pull request by innat**
fix https://github.com/innat/medic-ai/issues/61

## Summary by CodeRabbit

* **New Features**
  * Added SqueezeExcitation attention layer and ResNeXt50/ResNeXt101 model variants.

* **API Changes**
  * Public exports expanded to include layers, metrics, and losses.
  * Convolution block parameter renamed from name_prefix to name.

* **Improvements**
  * Global warning suppression removed so runtime diagnostics are visible.
  * Added runtime validation for activations, dropout ranges, and num_classes.

* **Documentation**
  * Added ResNeXt documentation and examples.

### CODE REVIEW

File: medicai/layers/attention.py [hunk 1]
```diff
@@ -98,3 +103,65 @@ def get_config(self):
...
+            spatial_dims=spatial_dims,
+            layer_type="conv",
+            filters=self.channels,
+            kernel_size=1,
+            activation="sigmoid",
+        )
+        pooled_shape_conv2 = self.conv1.compute_output_shape(pooled_shape_conv1)
+        self.conv2.build(pooled_shape_conv2)
+
+        super().build(input_shape)
```
[the remark this answers is not in the document]
Comment (innat) [reply medicai/layers/attention.py]: /gemini

Your suggestion is right when working with tensorflow backend. But in jax backend, each layer in build method needed to built manually, pesky, I know.

File: medicai/layers/attention.py [hunk 2]
```diff
@@ -98,3 +99,65 @@ def get_config(self):
...
```
[the remark this answers is not in the document]
Comment (innat) [reply medicai/layers/attention.py]: /gemini

It's for jax backend.

[the remark this answers is not in the document]
Comment (innat) [reply medicai/layers/attention.py]: /gemini 

This is for jax.
